cg/meta/workflow/analysis.py

```diff
@@ -495,10 +503,53 @@ def is_case_ready_for_analysis(self, case_id: str) -> bool:
             return False
         return True
 
+    def does_any_file_need_to_be_retrieved(self, case_id: str) -> bool:
+        """Checks whether we need to retrieve files from an external data location."""
+        if self.get_archive_location_for_case(case_id) == ArchiveLocations.PDC:
+            if self._is_flow_cell_check_applicable(
+                case_id
+            ) and not self.status_db.are_all_flow_cells_on_disk(case_id):
+                LOG.warning(f"Case {case_id} is not ready - all flow cells not present on disk.")
+                return True
+        else:
+            if not self.are_all_spring_files_present(case_id):
+                LOG.warning(f"Case {case_id} is not ready - some files are archived.")
+                return True
+        return False
+
     def prepare_fastq_files(self, case_id: str, dry_run: bool) -> None:
-        """Retrieves or decompresses fastq files if needed, upon which an AnalysisNotReady error
+        """Retrieves or decompresses Spring files if needed. If so, an AnalysisNotReady error
         is raised."""
-        self.ensure_flow_cells_on_disk(case_id)
+        self.ensure_spring_files_are_not_archived(case_id)
         self.resolve_decompression(case_id, dry_run=dry_run)
         if not self.is_case_ready_for_analysis(case_id):
             raise AnalysisNotReadyError("FASTQ file are not present for the analysis to start")
+
+    def ensure_spring_files_are_not_archived(self, case_id: str):
+        """Checks if any Spring files are archived and submits a job to retrieve any which are."""
+        if self.get_archive_location_for_case(case_id) == ArchiveLocations.PDC:
+            self.ensure_flow_cells_on_disk(case_id)
```

Comment:
Same here, `self.ensure_flow_cells_on_disk(case_id)` needs to be checked regardless of ArchiveLocation